app.go

```diff
@@ -603,6 +618,21 @@ func New(config ...Config) *App {
 	// Override colors
 	app.config.ColorScheme = defaultColors(app.config.ColorScheme)
 
+	// If the app is configured to use services,
+	// register a post shutdown hook to shutdown them after the server is closed.
+	if app.hasConfiguredServices() {
```

Comment:
Can we do the process in the `init` function ?
And can you outsource this whole block in a function called
`initServices`
i like short functions (CleanCode disciple)  and we blow up the `New` or `init` function with this whole block

Later

```go
func (app *App) init() *App {
	// lock application
	app.mutex.Lock()
	
	app.initServices()
	...
```

Reply:
@ReneWerner87 I applied your suggestion in [0a75429](https://github.com/gofiber/fiber/pull/3434/commits/0a75429d9a8168595c18dc3941e5a72616f2bd34). It' even greater, as it allowed me to write unit tests for that function easier. Thanks for the suggestion.

Reply:
> Can we do the process in the init function ?

I'm sorry but I don't get this part: what process do you refer? 🙏